writer/openapi/utils.go

```diff
@@ -15,23 +15,51 @@ type TypeInfo struct {
 }
 
 func openAPIType(p *schema.Property) (TypeInfo, error) {
-	if(p.Type == schema.Type_ARRAY) {
-		at, err := openAPIType_helper(p.GetArrayPrimitiveType(), p.GetArrayObjectType())
-		if(err != nil) {
-			return TypeInfo{}, nil
-		}
-		return TypeInfo{
-			openapi_type: "array",
-			array_type: &at,
-		}, nil
+	switch p.GetTypes().(type) {
+		case *schema.Property_ArrayType:
+			return openAPITypeArray(p.GetArrayType())
+		case *schema.Property_ObjectType:
+			return openAPITypeObject(p.GetObjectType())
+		case *schema.Property_Type:
+			return openAPITypePrimitive(p.GetType())
+		default:
+			return TypeInfo{}, fmt.Errorf("reached end of openapi type switch statement")
+	}
+}
+
+func openAPITypeArray(a *schema.ArrayType) (TypeInfo, error) {
+	switch a.GetArrayDetails().(type) {
+		case *schema.ArrayType_Type: 
+			at, err := openAPITypePrimitive(a.GetType())
+			if(err != nil) {
+				return TypeInfo{}, nil
+			}
+			return TypeInfo{
+				openapi_type: "array",
+				array_type: &at,
+			}, nil
+		case *schema.ArrayType_ObjectType:
+			ot, err := openAPITypeObject(a.GetObjectType())
+			if(err != nil) {
+				return TypeInfo{}, nil
+			}
+			ot.openapi_type = "array"
+			return ot, nil
+		
+		default:
+			return TypeInfo{} , fmt.Errorf("reached end of openAPITypeArray switch")
```

Comment:
```suggestion
			return TypeInfo{} , fmt.Errorf("OpenAPI type for %q not found", a.GetArrayDetails().(type))
```

Would be worth verifying what this actually shows up as.

Reply:
Same issue. Can't access the type information outside of a switch statement. Very strange Go-ism.